// @aws-sdk/client-translate/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextSizeLimitExceededException = exports.DetectedLanguageLowConfidenceException = exports.ServiceUnavailableException = exports.UnsupportedLanguagePairException = exports.InvalidFilterException = exports.UnsupportedDisplayLanguageCodeException = exports.ResourceNotFoundException = exports.TooManyTagsException = exports.TooManyRequestsException = exports.LimitExceededException = exports.InvalidRequestException = exports.InvalidParameterValueException = exports.InternalServerException = exports.ConflictException = exports.ConcurrentModificationException = void 0;
const TranslateServiceException_1 = require("./TranslateServiceException");
class ConcurrentModificationException extends TranslateServiceException_1.TranslateServiceException {
    name = "ConcurrentModificationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConcurrentModificationException = ConcurrentModificationException;
class ConflictException extends TranslateServiceException_1.TranslateServiceException {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConflictException = ConflictException;
class InternalServerException extends TranslateServiceException_1.TranslateServiceException {
    name = "InternalServerException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerException = InternalServerException;
class InvalidParameterValueException extends TranslateServiceException_1.TranslateServiceException {
    name = "InvalidParameterValueException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidParameterValueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidParameterValueException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidParameterValueException = InvalidParameterValueException;
class InvalidRequestException extends TranslateServiceException_1.TranslateServiceException {
    name = "InvalidRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRequestException = InvalidRequestException;
class LimitExceededException extends TranslateServiceException_1.TranslateServiceException {
    name = "LimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.LimitExceededException = LimitExceededException;
class TooManyRequestsException extends TranslateServiceException_1.TranslateServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
class TooManyTagsException extends TranslateServiceException_1.TranslateServiceException {
    name = "TooManyTagsException";
    $fault = "client";
    ResourceArn;
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.ResourceArn = opts.ResourceArn;
    }
}
exports.TooManyTagsException = TooManyTagsException;
class ResourceNotFoundException extends TranslateServiceException_1.TranslateServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class UnsupportedDisplayLanguageCodeException extends TranslateServiceException_1.TranslateServiceException {
    name = "UnsupportedDisplayLanguageCodeException";
    $fault = "client";
    Message;
    DisplayLanguageCode;
    constructor(opts) {
        super({
            name: "UnsupportedDisplayLanguageCodeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedDisplayLanguageCodeException.prototype);
        this.Message = opts.Message;
        this.DisplayLanguageCode = opts.DisplayLanguageCode;
    }
}
exports.UnsupportedDisplayLanguageCodeException = UnsupportedDisplayLanguageCodeException;
class InvalidFilterException extends TranslateServiceException_1.TranslateServiceException {
    name = "InvalidFilterException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidFilterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidFilterException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidFilterException = InvalidFilterException;
class UnsupportedLanguagePairException extends TranslateServiceException_1.TranslateServiceException {
    name = "UnsupportedLanguagePairException";
    $fault = "client";
    Message;
    SourceLanguageCode;
    TargetLanguageCode;
    constructor(opts) {
        super({
            name: "UnsupportedLanguagePairException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedLanguagePairException.prototype);
        this.Message = opts.Message;
        this.SourceLanguageCode = opts.SourceLanguageCode;
        this.TargetLanguageCode = opts.TargetLanguageCode;
    }
}
exports.UnsupportedLanguagePairException = UnsupportedLanguagePairException;
class ServiceUnavailableException extends TranslateServiceException_1.TranslateServiceException {
    name = "ServiceUnavailableException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceUnavailableException = ServiceUnavailableException;
class DetectedLanguageLowConfidenceException extends TranslateServiceException_1.TranslateServiceException {
    name = "DetectedLanguageLowConfidenceException";
    $fault = "client";
    Message;
    DetectedLanguageCode;
    constructor(opts) {
        super({
            name: "DetectedLanguageLowConfidenceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DetectedLanguageLowConfidenceException.prototype);
        this.Message = opts.Message;
        this.DetectedLanguageCode = opts.DetectedLanguageCode;
    }
}
exports.DetectedLanguageLowConfidenceException = DetectedLanguageLowConfidenceException;
class TextSizeLimitExceededException extends TranslateServiceException_1.TranslateServiceException {
    name = "TextSizeLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TextSizeLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TextSizeLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.TextSizeLimitExceededException = TextSizeLimitExceededException;
